newIDE/app/src/EventsSheet/ParameterFields/VariableField.js

```diff
@@ -297,6 +302,38 @@ export default React.forwardRef<Props, VariableFieldInterface>(
       !errorText &&
       value;
 
+    const openVariableEditor = React.useCallback(
+      () => {
+        if (!onOpenDialog) {
+          return;
+        }
+        // Access to the input directly because the value
+        // may not have been sent to onChange yet.
+        const fieldCurrentValue = field.current
+          ? field.current.getInputValue()
+          : value;
+        const isRootVariableDeclared =
+          variablesContainers &&
+          variablesContainers.some(variablesContainer =>
+            variablesContainer.has(getRootVariableName(fieldCurrentValue))
+          );
+
+        if (!field.current) {
+          onOpenDialog({
+            variableName: value,
+            shouldCreate: false,
+          });
+          return;
+        }
```

Comment:
There seems to be extra computation done for nothing if field.current is null and extra checks that can be avoided:
```suggestion
        if (!field.current) {
          onOpenDialog({
            variableName: value,
            shouldCreate: false,
          });
          return;
        }
        
        const fieldCurrentValue = field.current.getInputValue();
        const isRootVariableDeclared =
          variablesContainers &&
          variablesContainers.some(variablesContainer =>
            variablesContainer.has(getRootVariableName(fieldCurrentValue))
          );
```

Reply:
I forgot to remove a `if`.
The field ref always exist when the action from the autocompletion is selected.